examples/unimp_arxiv.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import os
+
+import torch
+from ogb.nodeproppred import PygNodePropPredDataset
+
+import torch_geometric.transforms as T
+from torch_geometric.nn import TransformerConv
+
+logger = logging.getLogger(__file__)
+logging.basicConfig(level=logging.INFO)
+
+# Get dataset
+root = os.path.join("..", "data", "OGB")
+dataset = PygNodePropPredDataset(
+    "ogbn-arxiv",
+    root,
+    transform=T.Compose([T.ToUndirected()]),
+)
+data = dataset[0]
+y_vocab_size = 40
+
+# Form masks for labels: mask gives labels being predicted
+split_idx = dataset.get_idx_split()
+train_mask = torch.zeros(data.x.size(0))
+train_mask = train_mask.scatter_(0, split_idx["train"], 1).type(torch.bool)
+
+valid_mask = torch.zeros(data.x.size(0))
+valid_mask = valid_mask.scatter_(0, split_idx["valid"], 1).type(torch.bool)
+
+test_mask = torch.zeros(data.x.size(0))
+test_mask = test_mask.scatter_(0, split_idx["test"], 1).type(torch.bool)
+
+# Model parameters
+inner_dim = 16
+heads = 2
+
+
+class UnimpNet(torch.nn.Module):
+    def __init__(self, feature_size: int, label_vocab: int, inner_dim: int,
+                 heads: int):
+        super().__init__()
+
+        self.label_embedding = torch.nn.Embedding(label_vocab, feature_size)
+        self.conv = TransformerConv(feature_size, inner_dim // heads,
+                                    heads=heads)
+        self.linear = torch.nn.Linear(inner_dim, label_vocab)
+
+    def forward(
+        self,
+        x: torch.Tensor,
+        y: torch.Tensor,
+        edge_index: torch.Tensor,
+        label_mask: torch.Tensor,
+    ):
+        x = torch.clone(x)
+        x[~label_mask] = x[~label_mask] + self.label_embedding(
```

Comment:
```suggestion
        x[~label_mask] += self.label_embedding(
```
should work as well, right?

Reply:
makes sense... I was actually trying to avoid the clone which felt like a bit of a hack (to be honest I didn't fully understand what was going on here, but it seemed like x was being free'd after the backwards pass because I'm modifying it in place).

Reply:
I don't think we can avoid the `clone` here since it looks like PyTorch does not expose a `masked_add` function :(

Reply:
I changed it to use a scatter and a zero vector.. but it might be worse 😓